### ISSUE TITLE
Facebook Login Not working for custom user model

### ISSUE THREAD

**Issue by w3gw** [python-social-auth/social-app-django#281]
ValueError at /social-login/complete/facebook/
User Must have email address
C:\Users\....\accounts\models.py in create_user, l

Request Method: | GET
-- | --
http://localhost:8000/social-login/complete/facebook/?granted_scopes=public_profile&denied_scopes&code=AQCvX2pvLJGjjsX0rtmOKLquom4sUZd1zTetQxj9r2fJJMvO6v8cpjGa-F9FIE6r5cOrykrLpgcIQS6wsflSyyfk6z7_v1f0hcvlSDLRVdf2t2-whLkzgBq2xiaQOqg29kS4ZYIuekWlW8hKFdWsoI-avI6gx7j9hoNh97Jej7sY2aezf4B4i0BGJrw-CFS5TCX8DpA7XmC_vhyOSK1wgfMlh3hoqTsImT1m08yBndEPPUOHUJkWd1hvuR_5kYX1gmmS1vgd5oZzy5Rpq2K80hFpoCvRkhhayiAfvD9JtHokSMD2Mzq3A5FtVrxIBg0Sx0GvDUeKPNEQC2VK3h4JjIX-&state=ty3opZX0HMfizMfJ3OJ83kxCup5X9C0g
2.2

**Comment by actlikewill** [comment]
I am having the same issue. Inspecting the details in the pipeline shows that the email address is an empty string so it cannot create a new user. I have added these settings in settings.py but it still doesn't work. 

```
SOCIAL_AUTH_FACEBOOK_SCOPE = ['email']
SOCIAL_AUTH_FACEBOOK_PROFILE_EXTRA_PARAMS = {'fields': 'id, name, email' }
SOCIAL_AUTH_FACEBOOK_SCOPE = ['email']
FACEBOOK_EXTENDED_PERMISSIONS = ['email']
```

Did you manage to solve this? I am thinking that this should be solved on facebook's end by getting a token with the email embedded in it but I am not sure.

**Comment by w3gw** [comment]
Yes I solved it. It was showing me that error because email attribute inside my model is required and it was not getting it from Facebook. so  I added this to my settings.py file and it worked.
...

SOCIAL_AUTH_FACEBOOK_SCOPE = ['email', 'user_link']  
SOCIAL_AUTH_FACEBOOK_PROFILE_EXTRA_PARAMS = {        
  'fields': 'id, name, email, link '
}

SOCIAL_AUTH_FACEBOOK_EXTRA_DATA = [                 
    ('name', 'name'),
    ('email', 'email'),
    ('picture', 'picture'),
    ('link', 'profile_url'),
]
may be you can try it
